File: Fenicia.Auth/Domains/Subscription/GetUserProfile/GetUserProfileResponse.cs
namespace Fenicia.Auth.Domains.Subscription.GetUserProfile;

public class GetUserProfileResponse
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public List<UserCompanyResponse> Companies { get; set; } = [];
    public List<UserSubscriptionResponse> Subscriptions { get; set; } = [];
}

File: Src/fenicia-web/src/types/auth/get-user-profile-response.ts
import { UserCompanyResponse } from "./user-company-response";
import { UserSubscriptionResponse } from "./user-subscription-response";

export type GetUserProfileResponse = {
    id: string;
    name: string;
    email: string;
    companies: UserCompanyResponse[];
    subscriptions: UserSubscriptionResponse[];
};
